src/user/user.service.ts

```diff
@@ -1,23 +1,49 @@
-import {Injectable} from '@nestjs/common';
+import {BadRequestException, Injectable} from '@nestjs/common';
 import {InjectRepository} from '@nestjs/typeorm';
 import {Repository} from 'typeorm';
 import {CreateUserDto} from './dto/createUserDto';
-import {User} from './user.entity';
+import {User} from './entities/user.entity';
+import {Profile} from '../profile/entities/profile.entity';
+
+import * as bcrypt from 'bcrypt';
+import {Err} from '../error';
 
 @Injectable()
 export class UserService {
   constructor(
     @InjectRepository(User)
     private readonly userRepository: Repository<User>,
-  ) {}
+    @InjectRepository(Profile)
+    private readonly profileRepository: Repository<Profile>,
+  ) {
+  }
+
+  async findUserByEmail(email: string) {
+    const user = await this.userRepository.findOne({
+      where: {
+        email,
+      },
+    });
+
+    return user;
+  }
 
   async createUser(createUserDto: CreateUserDto) {
+    const user = await this.findUserByEmail(createUserDto.email);
+
+    if (user) throw new BadRequestException(Err.EXISTING_USER);
+
+    const password = await bcrypt.hash(createUserDto.password, 12);
+
     const userData = await this.userRepository.save({
```

Comment:
```suggestion
    const createdUser = await this.userRepository.save({
```
`createdUser`로 변경하면 어떨까요?